board/fischertechnik/TXT/rootfs/opt/ftc/apps/system/display/display.py

```diff
@@ -49,6 +52,7 @@ def on_calibrate_touchscreen(self):
         self.mainWindow = TouchBaseWidget()
         self.mainWindow.show()
         old_window.close()
+        subprocess.run(["/sbin/set-touchscreen-calibration-reset-flag"])
```

Comment:
Now that you don't need sudo anymore here, why do you still fire off a subprocess just to create an empty file?

Reply:
Because the app is a wrapper around some scripts. If I change the system later, I don´t need to fix the app. It will just continue work

Reply:
But if you want, I can delete the scripts

Reply:
But then I need to change the boot-up script again

Reply:
> Because the app is a wrapper around some scripts. If I change the system later, I don´t need to fix the app.

That is a valid point in principle. But, as it is implemented now, you need to use three programs (`set-touchscreen-calibration-reset-flag`,  `ts_calibrate`, `unset-touchscreen-calibration-reset-flag`) in the right sequence, and two of them are just `touch` and `rm` with a fixed file. And to make the whole flag mechanism usable at all, you need to tie it in with `/etc/init.d/S80touchscreen-calibration` as well, which already uses a plain `cp` to revert the broken calibration data.

If you want to go for a full calibration 'backend', go all the way and build a CLI util that accepts parameters such as 'calibrate' (creates the reset flag, runs `ts_calibrate`), 'reset' (checks if a flag exists, restores the default calibration if yes, and deletes the flag) and 'commit' (simply deletes the flag), and use this (and only this) utility from the GUI and the init script.

Or drop the set/reset scripts altogether (that is my recommendation, I think a full CLI script is overkill here), even if that introduces a slight redundancy in thre GUI and init scripts (both need to remove the flag file).  

But please don't leave it in the current, half-baked state.

Reply:
I will build this cli thing. This sounds very good